src/components/BookmarkButton/hooks/useBookmarkButton.ts

```diff
@@ -0,0 +1,136 @@
+import { useSession, useSupabaseClient } from '@supabase/auth-helpers-react'
+import { useWallet } from 'hooks/Wallet'
+import { useWalletSignIn } from 'hooks/useWalletSignIn'
+import { PV } from 'models/pv'
+import { useCallback, useEffect, useState } from 'react'
+import { Database } from 'types/database.types'
+import { getSubgraphIdForProject } from 'utils/graph'
+import { emitErrorNotification } from 'utils/notifications'
+
+/**
+ * Hook to control the bookmark button.
+ * @param projectId The projectId of the project to bookmark.
+ * @param pv The PV of the project to bookmark.
+
+ * @returns The loading state, the bookmarked state, and the bookmark function.
+ * @example const { loading, isBookmarked, bookmark } = useBookmarkButton({ project: '1-17' })
+ */
+export const useBookmarkButton = ({
+  projectId,
+  pv,
+}: {
+  projectId: number
+  pv: PV
+}) => {
+  const [loading, setLoading] = useState<boolean>(false)
+  const [isBookmarked, setIsBookmarked] = useState<boolean>(false)
+
+  const wallet = useWallet()
+  const signIn = useWalletSignIn()
+  const session = useSession()
+  const supabase = useSupabaseClient<Database>()
+
+  const project = getSubgraphIdForProject(pv, projectId)
+
+  const getIsBookmarked = useCallback(
+    async ({ project, userId }: { project: string; userId: string }) => {
+      const { data, error } = await supabase
+        .from('user_bookmarks')
+        .select('*')
+        .eq('user_id', userId)
+        .eq('project', project)
+
+      if (error) {
+        console.error(error)
+        return
+      }
+
+      return data.length > 0
```

Comment:
**suggestion** kinda inconsequential at our scale but a `count` query might be better here:

https://stackoverflow.com/questions/65612167/how-to-get-count-in-supabase might work?